admin/RELEASING.md

```diff
@@ -0,0 +1,18 @@
+# How-to release `rustls-platform-verifier`
+
+This document records the steps to publish new versions of the crate since it requires non-trivial preperation and ordering
+that needs to be remembered due to the Android component's distribution.
+
+## Steps
+
+1. Update main crate'a version in `rustls-platform-verifier/Cargo.toml`, and in any additional places.
+2. If any non-test changes have been made to the `android` directory since the last release:
+    1. Update Android artifact version in `android-release-support/Cargo.toml`
+    2. Bump dependency version of the Android support crate in `rustls-platform-verifier/Cargo.toml` to match the new one
+    3. Commit version increase changes on the release branch
+    4. Run `ci/package_android_release.sh` in a UNIX compatible shell
+    5. (Optional) `cargo publish -p rustls-platform-verifier-android --dry-run`
```

Comment:
When I run this immediately after `ci/package_android_releas.sh` it bails from a dirty worktree:
```
❯ cargo publish -p rustls-platform-verifier-android --dry-run
    Updating crates.io index
error: 4 files in the working directory contain changes that were not yet committed into git:

maven/rustls/rustls-platform-verifier/0.1.0/rustls-platform-verifier-0.1.0.pom
maven/rustls/rustls-platform-verifier/0.1.0/rustls-platform-verifier-0.1.0.aar
maven/rustls/rustls-platform-verifier/maven-metadata-local.xml
maven/pom.xml

to proceed despite this and include the uncommitted changes, pass the `--allow-dirty` flag
```

Is the releaser supposed to use `--alow-dirty` since the include directive packages these files for a real release?

Reply:
They shouldn't need to, no. I'm not sure what the difference is yet but the new gitignore rules _should_ be preventing these from making the worktree dirty. Instead, the `include` directive in the Android support crate will make sure to grab them despite not being part of Git and remove the need for `--allow-dirty`.

Reply:
>  Running the script doesn't change git status for me locally.

It's the same for me FWIW. 

> They shouldn't need to, no. I'm not sure what the difference is yet but the new gitignore rules should be preventing these from making the worktree dirty.

It sounds like this might be intentional behaviour on the part of `cargo`: https://github.com/rust-lang/cargo/issues/12294

Reply:
I will try to investigate this more, but do you think this is a blocker? The Android-related artifacts are easily reproducible so my hope is that anyone who is less comfortable with "binary" artifacts being included in a crate's contents would be able to checkout the repository from the Git tag, regenerate these artifacts and assert their checksums match.

Reply:
I don't think it needs to be considered a blocker :+1: